nav2_bringup/bringup/params/nav2_params.yaml

```diff
@@ -90,7 +90,8 @@ controller_server:
     min_y_velocity_threshold: 0.5
     min_theta_velocity_threshold: 0.001
     progress_checker_plugin: "progress_checker"
-    goal_checker_plugin: "goal_checker"
+    goal_checker_plugins: ["goal_checker", "precise_goal_checker"]
+    current_goal_checker: "goal_checker" # it his param is not specified "goal_checker" is selected by default
```

Comment:
`default_` should be a more appropriate prefix than `current`, also please replace the comment.

Reply:
Well, this pull request was an initial contact to show how it works (we can change many things).

Concerning replacing "current" with "default", I would agree if it was the "default goal checker", but in fact, that parameter is being used as a "dynamic reconfigure" parameter so that it is used to select the "current goal checker". It may vary in the future (it is not just the initial default goal checker).

Maybe you  disagree with the mechanism of selecting goal checker via ros2 parameters.

Reply:
Lets discuss this in the comment below. When I made that request, I did not realize you were trying to use the parameter system dynamically. I don't think there's anything technically wrong about that strategy, but philosophically I don't think its going to have the best outcome for your needs. 

Also, ROS2 has a concept of parameter update event callbacks. In this situation, you'll polling for parameter updates. If we were to go in this direction (which I'm not sure is the right move), it should be event based.